### DISCUSSION TITLE
upstream to the mainline

### DISCUSSION THREAD

**Discussion by olumolu** [linux-surface/linux-surface#1581] [General]
Will this patches upstream to the mainline so we dont need to reinstall kernel we can have a out of the box. Experiance on serface devices thanks

**Comment by qzed** [comment]
A lot of things are already upstream. The biggest things not upstream are touchscreen and cameras (if they work at all). We aim to upstream these things as well once they are ready.

**Reply by olumolu** [reply]
Thanks.

**Comment by Headcrabed** [comment]
@qzed Seems intel THC driver is already mainlined, what is still needed to get mainlined for touch screen/stylus?

**Reply by qzed** [reply]
@quo and @StollD can probably answer that better than me. As far as I understand, the upstream driver only supports some newer versions of the hardware, so for now we still need the downstream driver for most models.

**Reply by quo** [reply]
Upstream driver only supports MTL and up (so only SP10, SL6 and up). See https://github.com/quo/ithc-linux/issues/29
